src/integrationTest/sam.test.ts

```diff
@@ -343,10 +335,45 @@ describe('SAM Integration Tests', async () => {
                         )
                     })
 
-                    await vscode.commands.executeCommand<any>(
-                        codeLens.command!.command,
-                        ...codeLens.command!.arguments!
-                    )
+                    // const tempFolder = await makeTemporaryToolkitFolder()
+                    const fakeContext = await FakeExtensionContext.getFakeExtContext()
```

Comment:
The integration tests (under `src/integrationTest`) aren't intended to use any mocks/fakes/stubs. They should be exercising the actual extension as best they can.

In the `invokes the Debug Local CodeLens` test (prob needs a new name), we should be able to craft a json payload that looks like it came from a launch.json entry, and pass that in to `vscode.debug.startDebugging`. This should invoke the running extension's debug handling functionality. The debugSessionStartedAndStoppedPromise code can then check for debug session start and stop based on the `name` field provided within the json payload. One of these would be required per runtime, running against the sam app created in "Starting with a newly created ${scenario.runtime} SAM Application...".

Reply:
Invoking `startDebugging` directly, skips much of the logic, though.

Reply:
My understanding is that `startDebugging` would exercise `SamDebugConfigProvider::resolveDebugConfiguration`. Is that correct?

I wouldn't think the same test would exercise `provideDebugConfigurations`, in case this is what you are referring to. That would need separate test(s).

Reply:
> `startDebugging` would exercise `SamDebugConfigProvider::resolveDebugConfiguration`. Is that correct?

`startDebugging` is called much later, in the current implementation (see [jmk-rm-startDebugging](https://github.com/aws/aws-toolkit-vscode/tree/jmk-rm-startDebugging) for work towards removing it).

 I've updated the PR with a new approach which tests the full codepath:

1. add a config to launch.json via `launchConfig.addDebugConfiguration(testConfig)`
2. invoke F5 via `vscode.commands.executeCommand('workbench.action.debug.start')`